src/figuresClasses.ts

```diff
@@ -1,19 +1,93 @@
-export interface Figure {
-
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle'
 }
 
-export class Triangle {
+type Color = 'red' | 'green' | 'blue';
 
+function roundingArea(area: number): number {
+  return Math.floor(area * 100) / 100;
 }
 
-export class Circle {
+export interface Figure {
+  shape : Shape,
+  color: Color,
+  getArea(): number,
+}
+
+export class Triangle implements Figure {
+  readonly shape = Shape.Triangle;
+
+  constructor(
+    public color: Color,
+    public a: number = 0,
+    public b: number = 0,
+    public c: number = 0,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
```

Comment:
This and all similar checks can be done with a helper function and array method (some | every) on arguments list.